Add norebalance flag

tchaikov opened the pull request:
do we have any existing tests so I can follow to test this flag?

liewegas commented on doc/release-notes.rst:
```diff
@@ -209,6 +209,7 @@ Notable Changes
 * msgr: fix RESETSESSION bug (#10080 Greg Farnum)
 * objectstore: deprecate collection attrs (Sage Weil)
 * osd, mon: add checksums to all OSDMaps (Sage Weil)
+* osd, mon: honor new 'norebalance' osdmap flag
```
drop this hunk.  the release-notes are written after the next release is frozen; this is the last release. we don't know when this will appear yet..

liewegas commented:
the rest looks right to me.  please add a test to qa/workunits/cephtool/test.sh (to verify command parsing).

also, this might not be completely accurate: imagine a pg wants [1,2,3] and is stored on [1,4] with a partial instance on 2. i think (?) it will backfill to both 2 and 3, while technically it should probably only backfill to 2.  but i don't think it really matters... @athanatos ?

athanatos commented:
I think that particular wrinkle we can leave for later.

tchaikov commented:
@liewegas and @athanatos, thanks for your review. i 

- added the flag to the test of `mon_osd`
- and put the `norebalance` flag right after `nobackfill` in code, as they are more related.
- removed the change in the release-notes.

wondering why loic's bot failed to pick up my pull requests...
